src/python/pants/engine/process_test.py

```diff
@@ -411,3 +412,102 @@ def test_workspace_execution_support() -> None:
     assert result3.stderr.decode() == "this-goes-to-stderr\n"
     snapshot = rule_runner.request(Snapshot, [result3.output_digest])
     assert snapshot.files == ("capture-this-file",)
+
+
+@pytest.mark.parametrize(
+    "concurrency",
+    [
+        ProcessConcurrency.exactly(1),
+        ProcessConcurrency.exactly(2),
+        ProcessConcurrency.exclusive(),
+    ],
+)
+def test_concurrency(rule_runner: RuleRunner, concurrency: ProcessConcurrency) -> None:
+    test_description = f"concurrency-test-{concurrency.kind}-{concurrency.min}-{concurrency.max}"
+    process = Process(
+        argv=("/bin/echo", test_description),
+        concurrency=concurrency,
+        description=test_description,
+    )
+    result = rule_runner.request(ProcessResult, [process])
+    assert result.stdout.decode() == test_description + "\n"
+    assert result.stderr == b""
+
+
+@pytest.mark.parametrize(
+    "concurrency",
+    [
+        ProcessConcurrency.range(1, min=1),
+        ProcessConcurrency.range(max=2),
+        ProcessConcurrency.range(max=2, min=1),
+        # Values larger than num cores still work (they get clamped to num cores)
+        ProcessConcurrency.range(max=10000),
+        ProcessConcurrency.range(min=100, max=200),
+    ],
+)
+def test_concurrency_range(rule_runner: RuleRunner, concurrency: ProcessConcurrency) -> None:
+    test_description = f"concurrency-test-{concurrency.kind}-{concurrency.min}-{concurrency.max}"
```

Comment:
The argv needed to be unique, not just `concurrency` otherwise the process wouldn't actually run. I know it wasn't running because the range ones will err in bounded.rs if they don't include `{pants_concurrency}` which they did not initially